src/index.js

```diff
@@ -12,7 +13,9 @@ ReactDOM.render(
   <React.StrictMode>
     <BrowserRouter>
       <DataProvider>
-        <App />
+        <AuthProvider>
```

Comment:
AuthProvider can put as a parent of DataProvider

Reply:
done, will be in next commit.